app/services/catalog/determine_task_relevancy.rb

```diff
@@ -53,5 +54,10 @@ def add_update_message(state, message)
       order_item.update_message(state, message)
       Rails.logger.send(state, message)
     end
+
+    def fail_order
+      order_item.update!(:state => "Failed")
+      Catalog::OrderStateTransition.new(order_item.order.id).process
```

Comment:
Realize this isn't really for this PR change, but do you know why `OrderStateTransition` takes an Order ID instead of just passing the object (which the service class immediately looks up)?

Also, other callers of `OrderStateTransition` are passing `order_id` (instead of `order.id`) I assume to avoid the `Order` object lookup here which is also going to happen in the service class.

Reply:
@gmcculloug That is part of our service object definition they always get passed an id and then they look up the object. So  when the service object is called from the controller we only have the params[:id], when the service object is called from other service objects we still stick to the same protocol of passing in ids.

Reply:
Makes total sense, I guess maybe just passing `object_id` like the other callers.

https://github.com/RedHatInsights/catalog-api/search?q=Catalog%3A%3AOrderStateTransition&unscoped_q=Catalog%3A%3AOrderStateTransition

Reply:
@eclarizio Does my request above make sense to change to `order_id`?  I think that's the only thing holding up this PR from my opinion.  If not we can merge.

Reply:
@gmcculloug Yeah we could change it, when I refactored the task stuff with the `DetermineTaskRelevancy` model, I ran into a similar issue where I had already looked up the order or order_item and didn't want to keep passing the ID around to look them up again so I passed the actual object. But here you're right, we haven't actually looked up the `order`, so I'll change it to just `order_id`.

Reply:
This thread got me thinking the other day and I verified the `Order.find_by(:id => object) with either the ID or the order object returns the expected order object.

```yaml
irb(main):001:0> order = Order.last
=> #<Order id: 107, user_id: nil, state: "Completed", created_at: "2019-05-02 19:00:59", order_request_sent_at: "2019-05-02 19:01:11", completed_at: nil, updated_at: "2019-05-02 19:02:07">

irb(main):002:0> Order.find_by(:id => order.id)
=> #<Order id: 107, user_id: nil, state: "Completed", created_at: "2019-05-02 19:00:59", order_request_sent_at: "2019-05-02 19:01:11", completed_at: nil, updated_at: "2019-05-02 19:02:07">

irb(main):003:0> Order.find_by(:id => order)
=> #<Order id: 107, user_id: nil, state: "Completed", created_at: "2019-05-02 19:00:59", order_request_sent_at: "2019-05-02 19:01:11", completed_at: nil, updated_at: "2019-05-02 19:02:07">
```
So it does not look like it matters which you pass to the service class initialization.  But to be honest I don't know why this works in Rails so hopefully it is not magic that gets removed from the framework at some point.